subgraphs/mux-protocol/src/mappings/gmxHandlers.ts

```diff
@@ -0,0 +1,262 @@
+} from "../utils/numbers";
+import { handleUpdatePositionEvent } from "./handlers";
+
+export function handleCreateProxyForGmx(event: CreateProxy): void {
+  const proxyMap = new _ProxyMap(event.params.proxy);
+  proxyMap.positionId = event.params.owner;
+  proxyMap.save();
+}
+
+export function handleIncreasePosition(event: IncreasePosition): void {
+  handleGmxUpdatePositionEvent(
+    event,
+    event.params.key,
+    event.params.account,
+    event.params.collateralToken,
+    event.params.collateralDelta,
+    event.params.indexToken,
+    event.params.sizeDelta,
+    event.params.price,
+    event.params.isLong,
+    EventType.CollateralIn,
+    BIGINT_ZERO
+  );
+}
+
+export function handleDecreasePosition(event: DecreasePosition): void {
+  handleGmxUpdatePositionEvent(
+    event,
+    event.params.key,
+    event.params.account,
+    event.params.collateralToken,
+    event.params.collateralDelta,
+    event.params.indexToken,
+    event.params.sizeDelta,
+    event.params.price,
+    event.params.isLong,
+    EventType.CollateralOut,
+    BIGINT_ZERO
+  );
+}
+
+export function handleLiquidatePosition(event: LiquidatePosition): void {
+  handleGmxUpdatePositionEvent(
+    event,
+    event.params.key,
+    event.params.account,
+    event.params.collateralToken,
+    event.params.collateral,
+    event.params.indexToken,
+    event.params.size,
+    event.params.markPrice,
+    event.params.isLong,
+    EventType.Liquidated,
+    event.params.realisedPnl
+  );
+}
+
+export function handleGmxUpdatePositionEvent(
+  event: ethereum.Event,
+  positionKey: Bytes,
+  accountAddress: Address,
+  collateralTokenAddress: Address,
+  collateralDelta: BigInt,
+  indexTokenAddress: Address,
+  sizeDelta: BigInt,
+  indexTokenPrice: BigInt,
+  isLong: boolean,
+  eventType: EventType,
+  liqudateProfit: BigInt
+): void {
+  // For every trading being routed to GMX, MUX protocol creates a proxy address for the trader.
+  // If the trader address from GMX contract does not exist in proxy address list in MUX protocol, the trading does not come from MUX protocol.
+  const proxyMap = _ProxyMap.load(accountAddress);
+  if (!proxyMap) {
+    return;
+  }
+
+  const pool = getOrCreateLiquidityPool(
+    event,
+    event.address,
+    GMX_POOL_NAME,
+    GMX_POOL_SYMBOL
+  );
+  takeSnapshots(event, pool);
+
+  const account = getOrCreateAccount(event, pool, accountAddress);
+  incrementAccountEventCount(event, pool, account, eventType, sizeDelta);
+  incrementProtocolEventCount(event, eventType, sizeDelta);
+
+  const indexToken = getOrCreateToken(event, indexTokenAddress);
+  updateTokenPrice(
+    event,
+    indexToken,
+    convertToDecimal(indexTokenPrice, GMX_PRICE_DECIMALS)
+  );
+  const sizeUSDDelta = convertToDecimal(sizeDelta, GMX_PRICE_DECIMALS);
+  const collateralToken = getOrCreateToken(event, collateralTokenAddress);
+  const collateralUSDDelta = convertToDecimal(
+    collateralDelta,
+    GMX_PRICE_DECIMALS
+  );
```

Comment:
GMX amounts for position collateral and size are USD denominated?

Reply:
Yes. The position collateral and size from GMX event are denominated in USD and just need to do some decimal conversion before use.